gfx/drivers_shader/slang_process.cpp

```diff
@@ -0,0 +1,406 @@
+      unsigned               version,
+      const semantics_map_t* semantics_map,
+      pass_semantics_t*      out)
+{
+   Compiler*          vs_compiler = NULL;
+   Compiler*          ps_compiler = NULL;
+   video_shader_pass& pass        = shader_info->pass[pass_number];
+   glslang_output     output;
+
+   if (!glslang_compile_shader(pass.source.path, &output))
+      return false;
+
+   if (!slang_preprocess_parse_parameters(output.meta, shader_info))
+      return false;
+
+   pass.source.string.vertex   = NULL;
+   pass.source.string.fragment = NULL;
+
+   try
+   {
+      ShaderResources vs_resources;
+      ShaderResources ps_resources;
+      string          vs_code;
+      string          ps_code;
+
+      if (dst_type == RARCH_SHADER_HLSL || dst_type == RARCH_SHADER_CG)
+      {
+         vs_compiler = new CompilerHLSL(output.vertex);
+         ps_compiler = new CompilerHLSL(output.fragment);
+      }
+      else
+      {
+         vs_compiler = new CompilerGLSL(output.vertex);
+         ps_compiler = new CompilerGLSL(output.fragment);
+      }
+
+      vs_resources = vs_compiler->get_shader_resources();
+      ps_resources = ps_compiler->get_shader_resources();
+
+      if (dst_type == RARCH_SHADER_HLSL || dst_type == RARCH_SHADER_CG)
+      {
+         CompilerHLSL::Options options;
+         CompilerHLSL*         vs = (CompilerHLSL*)vs_compiler;
+         CompilerHLSL*         ps = (CompilerHLSL*)ps_compiler;
+         options.shader_model     = version;
+         vs->set_options(options);
+         ps->set_options(options);
+
+         std::vector<HLSLVertexAttributeRemap> vs_attrib_remap;
+#if 0
+         /* remaps vertex shader output too so disable for now */
+         vs_attrib_remap.push_back({ 0, "POSITION" });
+         vs_attrib_remap.push_back({ 1, "TEXCOORD" });
+#endif
+         /* not exactly a vertex attribute but this remaps
+          * float2 FragCoord :TEXCOORD# to float4 FragCoord : SV_POSITION */
+         std::vector<HLSLVertexAttributeRemap> ps_attrib_remap;
+         for (Resource& resource : ps_resources.stage_inputs)
+         {
+            if (ps->get_name(resource.id) == "FragCoord")
+            {
+               uint32_t location = ps->get_decoration(resource.id, spv::DecorationLocation);
+               ps_attrib_remap.push_back({ location, "SV_Position" });
+            }
+         }
+         VariableTypeRemapCallback ps_var_remap_cb =
+               [](const SPIRType& type, const std::string& var_name, std::string& name_of_type) {
+                  if (var_name == "FragCoord")
+                     name_of_type = "float4";
+               };
+         ps->set_variable_type_remap_callback(ps_var_remap_cb);
+
+         vs_code = vs->compile(vs_attrib_remap);
+         ps_code = ps->compile(ps_attrib_remap);
+      }
+      else if (shader_info->type == RARCH_SHADER_GLSL)
+      {
+         CompilerGLSL::Options options;
+         CompilerGLSL*         vs = (CompilerGLSL*)vs_compiler;
```

Comment:
How old is the SPIRV-Cross version in the repo? Should be the very latest if anything.

Reply:
I just updated it this week, the old one didn't have CompilerHLSL in the first place.